### DISCUSSION TITLE
Having trouble with completions working

### DISCUSSION THREAD

**Discussion by bbarker** [scalameta/metals#2948] [General]
Hello,

I'm not sure if this is a user error or something else. I've not used Metals much (maybe once before on a different system).

When I try to use code completion for methods and members (the only type of completion I've tried), nothing seems to happen. Note I'm just expecting a list to pop up after typing a ".". Maybe I should be doing something else. Note also that error reporting does work as shown by red squigglies when the wrong method name is used:
![vscode_metals_no_completion](https://user-images.githubusercontent.com/916366/125109342-39e3dd80-e0b1-11eb-90e7-c0c19646d339.gif)

Metals Doctor seems to report everything is well:
![image](https://user-images.githubusercontent.com/916366/125109122-effaf780-e0b0-11eb-9895-66c763ffe8f0.png)

I have attached metals.log as well.
[metals.log](https://github.com/scalameta/metals/files/6792764/metals.log)

Thanks for any suggestions!

**Comment by bbarker** [comment]
### Some more info

This is a Scala 3.0.0 project. I just uploaded it [to GitHub](https://github.com/bbarker/zionomicon-exs) ([initial commit](https://github.com/bbarker/zionomicon-exs/tree/b842228dd33c565557226350b0487ce5ede79f7e)).

Also, here is the metals version info:

```
$ metals --version
metals 0.10.4

# Note:
#   Supported Scala versions:
#     Scala 3: 3.0.0, 3.0.0-RC1, 3.0.0-RC2, 3.0.0-RC3
#     Scala 2:
#       2.11.12
#       2.12.14, 2.12.13, 2.12.12, 2.12.11, 2.12.10, 2.12.8, 2.12.9
#       2.13.6, 2.13.3, 2.13.4, 2.13.5, 2.13.0, 2.13.1, 2.13.2
```

VS Code is at 1.58

This is on OS X Catalina (10.15.7)

**Comment by bbarker** [comment]
After some more investigation, we found that it is related to this dependency (removing it enables completion). Note that I'm not directly making use of this dependency in the codebase as yet:

```scala
("io.github.kitlangton" %% "zio-magic"           % "0.3.5").cross(CrossVersion.for3Use2_13)
```

**Comment by tgodzik** [comment]
Thanks for the question! It seems that there is an issue with the dependencies:
```
Exception in thread "pool-4-thread-15" dotty.tools.dotc.core.TypeError: Toplevel definition TagK3 is defined in
  /Users/bbarker/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/dev/zio/izumi-reflect_2.13/1.1.2/izumi-reflect_2.13-1.1.2.jar(izumi/reflect/package.class)
and also in
  /Users/bbarker/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/dev/zio/izumi-reflect_3/1.1.2/izumi-reflect_3-1.1.2.jar(izumi/reflect/Tags$package.class)
One of these files should be removed from the classpath.
```

That error is coming from the presentation compiler, it has two versions of the same library on classpath. It's likely that `"zio-magic" ` uses a different version of the library transitively.

Also, there might be some other issues when using Scala 2.13 library that heavily uses macros, but not sure if this is the case here.

I managed to get the workspace working, so this might be connected to some issue within your environment maybe? Not sure.
